src/utils_general/ghs.py

```diff
@@ -0,0 +1,120 @@
+    :param column_row_pairs: List of column & row tuples specifying
+        tiles to download.
+    :param country_iso3: Country ISO3 code.
+    :param output_dir: Path or string specifying output directory
+        for output saving.
+
+    :return: Mosaic raster
+    """
+
+    # URL paths for GHS download
+
+    URL_BASE = "https://cidportal.jrc.ec.europa.eu/ftp/jrc-opendata/GHSL"
+    GHS_URL = {
+        "SMOD": "GHS_SMOD_POP_GLOBE_R2019A"
+        "/GHS_SMOD_POP2015_GLOBE_R2019A_54009_1K/V2-0/tiles/"
+        "GHS_SMOD_POP2015_GLOBE_R2019A_54009_1K_V2_0_{column}_{row}.zip",
+        "POP": "GHS_POP_MT_GLOBE_R2019A"
+        "/GHS_POP_E2015_GLOBE_R2019A_54009_1K/V1-0/tiles/"
+        "GHS_POP_E2015_GLOBE_R2019A_54009_1K_V1_0_{column}_{row}.zip",
+    }
+
+    # Set filepaths
+    download_dir = os.path.join(output_dir, "ghs", "zip")
+    output_filepath = os.path.join(
+        output_dir,
+        "ghs",
+        "{country_iso3}_{ghs_type}_2015_1km_mosaic.tif".format(
+            country_iso3=country_iso3, ghs_type=ghs_type
+        ),
+    )
+
+    logger.info(f"Getting GHS data for {ghs_type}")
+
+    Path(download_dir).mkdir(parents=True, exist_ok=True)
+    for column, row in column_row_pairs:
+        zip_filename = os.path.join(
+            download_dir, f"{ghs_type}_2015_1km_{column}_{row}.zip"
+        )
+        utils.download_url(
+            f"{URL_BASE}/{GHS_URL[ghs_type].format(column=column, row=row)}",
+            zip_filename,
+        )
+        utils.unzip(zip_filename, download_dir)
+
+    # Make a mosaic
+    files_to_mosaic = [
+        rasterio.open(f)
+        for f in glob.glob(os.path.join(download_dir, f"*_{ghs_type}_*.tif"))
+    ]
+
+    logger.info(f"Making mosiac of {len(files_to_mosaic)} files")
+    mosaic, out_trans = merge(files_to_mosaic)
+    out_meta = files_to_mosaic[0].meta.copy()
+    out_meta.update(
+        {
+            "driver": "GTiff",
+            "height": mosaic.shape[1],
+            "width": mosaic.shape[2],
+            "transform": out_trans,
+        }
+    )
+
+    with rasterio.open(output_filepath, "w", **out_meta) as dest:
+        dest.write(mosaic)
+
+    logger.info(f"Wrote file to {output_filepath}")
+
+
+def classify_urban_areas(
+    polygons,
+    ghs_smod,
+    transform,
+    urban_min_class: int = 21,
+    urban_percent: float = 0.5,
+):
+    """
+    Classifies polygons as urban based on SMOD raster data.
+
+    :param shape: Shapefile to aggregate to.
+    :param ghs_smod: Array of GHS SMOD data.
+    :param transform: Transform for GHS SMOD data.
+    :param urban_min_class: Minimum raster value to consider as urban area.
+        Defaults to 21, corresponding to suburban or peri-urban grid cells.
+    :param urban_percent: Percent of raster cells classified as urban to
+        consider polygon as urban.
+    """
+
+    def urban_area(x):
+        return np.ma.mean(x >= urban_min_class) >= urban_percent
+
+    return zonal_stats(
```

Comment:
If you use (rio)xarray, you could also use our [own zonal stats function](https://github.com/OCHA-DAP/pa-anticipatory-action/blob/a1fefb4d34124393c87d50b0214887ae4dc36903/src/utils_general/raster_manipulation.py#L110). As we have sometimes seen that `rasterstats.zonal_stats` doesn't handle the coordinates fully correctly. 
However, I trust it handles it correctly in this case so fine with leaving it as it is!

Reply:
Will do it this way in the future! And can amend this if we come back to it.